### ISSUE TITLE
Running non-python tasks

### ISSUE THREAD

**Issue by tachyus-ryan** [argoproj-labs/hera-workflows#20]
For some steps in my workflows, I need to run a bash script or container image. Can I do this with hera, or is this meant for pure python workflows?

**Comment by flaviuvadan** [comment]
> For some steps in my workflows, I need to run a bash script or container image. Can I do this with hera, or is this meant for pure python workflows?

I kind of expected containers and bash to work but, to be _very_ honest, I have not tried it 😅  as there was never a need for it. However, I just did, and containers work fine because you can specify the `image` and `command`!

```python
from hera.v1.task import Task
from hera.v1.workflow import Workflow
from hera.v1.resources import Resources

# omitted workflow service definition, you can inject your own!
w = Workflow('fv-testing')

# no point in requesting a lot of resources for a print statement, essentially
r = Resources(min_cpu=0.1, min_mem='100Mi')

# notice the placeholder `lambda: _`, for now this is required because 
# func is a required parameter, by design. There's an opportunity here,
# maybe, to make it an `Optional[Callable]`
t = Task('cowsay', lambda: _, image='docker/whalesay', command=['cowsay', 'moo'], resources=r)
w.add_task(t)
w.submit()
```

Results:
<img width="368" alt="Screen Shot 2021-12-02 at 07 06 49" src="https://user-images.githubusercontent.com/18152652/144419010-2de9dcbb-ea83-4bf2-a820-d53e51998824.png">

We have something _similar_ going on internally at @dynotx! We use Horovod and we pass `horovodrun`, along whatever arguments we need (e.g number of GPUs), inside the `command`. Just a note - internally, Hera will parse out numbers from the `command` field into strings!

I've tried a bash script inside the `func` that is added to a `Task` but that did not go so well. The main problem is that calling a test script that defines the workflow using `python` will result in a `SyntaxError` 🙁 Here's what I tried:

```python
from hera.v1.task import Task
from hera.v1.workflow import Workflow
from hera.v1.resources import Resources

# VERSION 1
def script(name: str):
    echo "Hello, ${name}!"

# omitted workflow service definition, you can inject your own!
w = Workflow('fv-testing')

# no point in requesting a lot of resources for a print statement, essentially
r = Resources(min_cpu=0.1, min_mem='100Mi')

t = Task('bash', script, func_params=[{'name': 'world'}], resources=r, command=['bash'])
w.add_task(t)
w.submit()
```

Result:
```shell
(hera) ➜  hera-workflows git:(main) ✗ python test.py                                                                                                                                                                    4ms 
  File "test.py", line 5
    echo "Hello, ${name}!"
                         ^
SyntaxError: invalid syntax
```

Another thing I've tried is submitting the bash script via a parameter and executing it using:
```python
def run_script(script):
    import subprocess
    proc = subprocess.Popen(['bash', '-c', script],
                            stdout=subprocess.PIPE,
                            stderr=subprocess.PIPE,
                            stdin=subprocess.PIPE)
    stdout, stderr = proc.communicate()
    if proc.returncode:
        raise Exception(proc.returncode, stdout, stderr, script)
    print(stdout, stderr)

script = "echo \"Hello, world!\""
```

However, that resulted in deserialization problems because of quoting/unquoting... probably best, for now, is to submit bash script execution through containers.

**Comment by flaviuvadan** [comment]
Merged #22 and closing this for now. @tachyus-ryan I hope this helped! We should definitely reopen this if we can have further discussions 🙂
